Lesson18/duygu/script.js

```diff
@@ -0,0 +1,197 @@
+console.log([integersEx1, integersEx1V2]); // [[1, 2, 3, 4], [10, 2, 30, 4]]
+
+console.log("Exercise 1: ");
+console.log('----------------------------');
+// 2. Write a function that takes an array of integers and returns the average of all elements.
+//Input: [1, 2, 3, 4]
+// Output: 2.5
+const integersEx2 = [1, 2, 3, 4];
+
+function getAvgOfArray(array) {
+
+  let sum = 0;
+
+  for (let i = 0; i < array.length; i++) {
+    sum += array[i];
+
+  }
+  const averageArray = sum / array.length;
+  return averageArray;
+}
+
+console.log("Exercise 2:");
+console.log(getAvgOfArray(integersEx2));
+console.log('----------------------------');
+// 3. Write a function that takes an array of strings and concatenates them together adding spaces.
+// Input: ["Hello", "World", "!"]
+// Output: "Hello World !"
+
+let stringE3 = ["Hello", "World", "!"];
+// const spacialCharacter = ['.', ',', ':', ';', '!'];
+function concatStrings(array) {
+  return array.join(' ')
+}
+
+
+console.log("Exercise 3:");
+console.log(concatStrings(stringE3));
+console.log('----------------------------');
+// 4. Write a function that takes an array of integers and returns a new array with odd numbers only.
+// Input: [1, 2, 3, 4, 7, 10]
+// Output: [1, 3, 7]
+let arrayEx = [1, 2, 3, 4, 7, 10];
+
+function getOddNumbers(array) {
+  let oddNumbers = [];
+  for (let i = 0; i < array.length; i++) {
+    if (array[i] % 2 !== 0) {
+      oddNumbers.push(array[i]);
+    }
+  }
+  return oddNumbers;
+}
+console.log("Ex4");
+console.log(getOddNumbers(arrayEx));
+console.log('----------------------------');
+// 5. Write a function that takes an array of integers and returns a new array with even numbers only.
+// Input: [1, 2, 3, 4, 7, 10]
+// Output: [1, 2, 4, 10]
+const testArrayEx5 = [1, 2, 3, 4, 7, 10];
+
+function getEvenNumbers(array) {
+  let evenNumbers = [];
+  for (let i = 0; i < array.length; i++) {
+    if (array[i] % 2 === 0) {
+      evenNumbers.push(array[i]);
+    }
+  }
+  return evenNumbers;
+}
+
+console.log('Exercise 5: ');
+console.log(getEvenNumbers(testArrayEx5));
+console.log('----------------------------');
+// 6. Write a function that takes an array and a value, and returns a new array with all occurrences of the value removed.
+// Input: ([1, 2, 3, 4, 2], 2)
+// Output: [1, 3, 4]
+const testArrayEx6 = [1, 2, 3, 4, 2];
+
+function removeValueFromArray(value, array) {
+  let result = [];
+
+  for (let i = 0; i < array.length; i++) {
+    if (value !== array[i]) {
+      result.push(array[i]);
+    }
+    // result = [...result, array[i]]
+
+  }
+  return result;
+}
+
+console.log("exercise 6:");
+console.log(removeValueFromArray(2, testArrayEx6));
+console.log('----------------------------');
+// 7. Write a function that takes an object and returns the keys as an array.
+// Input: {name: "John", age: 30, city: "New York"}
+// Output: ["name", "age", "city"]
+const testObjEx7 = { name: "John", age: 30, city: "New York" };
+function getObjectKeys(obj) {
```

Comment:
As alternative, there's a built in function in JS: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/keys